als-actions/shared/src/main/scala/org/mulesoft/als/actions/hover/PatchedHover.scala

```diff
@@ -38,13 +51,62 @@ class PatchedHover(provider: VocabularyProvider) {
   private def buildTerm(meta: ValueType, key: String, dialectName: String): ValueType =
     ValueType(s"http://als.patched/#$dialectName/${meta.name}/$key")
 
-  private val dialectNames: Map[String, String] =
-    Map(
-      Raml10TypesDialect.dialect.id -> "raml10",
-      Raml08TypesDialect.dialect.id -> "raml08",
-      OAS20Dialect.dialect.id       -> "oas2",
-      OAS30Dialect.dialect.id       -> "oas3",
-      AsyncApi20Dialect.dialect.id  -> "asyncapi2"
-    )
+  private def indexAliasTerms(valueType: ValueType,
+                              aliases: Aliases,
+                              dialectName: String,
+                              metaName: String): VocabularyProvider = {
+    val generatedTerms: Seq[ClassTerm] = aliases.aliases
+      .map(alias => {
+        val aliasTerm = buildAliasTerm(dialectName, metaName, alias._1)
+        ClassTerm()
+          .withName(aliasTerm)
+          .withDisplayName(aliasTerm)
+          .withDescription(Raml10UsesKeyTerm.description)
+      })
+      .toSeq
+    val hoverProvider = provider.asInstanceOf[AlsVocabularyRegistry]
+    hoverProvider.index(valueType.ns.base, generatedTerms)
+    hoverProvider
+  }
+
+  private def buildAliasTerm(dialectName: String, metaName: String, name: String): String =
+    s"$dialectName/${metaName}/$name"
+
+  private def patches(bu: BaseUnit, dialectName: String, metaUri: Option[String], entry: YMapEntry): Seq[() => Unit] = {
+    metaUri
+      .map(
+        meta => {
+          val valueType = buildTerm(ValueType(meta), entry.key.toString, dialectName)
+          Seq(
+            () =>
+              bu.annotations
+                .find(classOf[Aliases])
+                .foreach(aliases => indexAliasTerms(valueType, aliases, dialectName, ValueType(meta).name)))
+        }
+      )
+      .getOrElse(Seq.empty)
+  }
 
+  private def init(seq: Seq[(BaseUnit, Dialect)], branch: YPartBranch, obj: AmfObject): Unit = {
+    provider match {
+      case _: AlsVocabularyRegistry if branch.parentEntry.nonEmpty => {
+        val metaUri = obj.metaURIs.headOption
+        seq.foreach(t =>
+          patches(t._1, dialectNames.getOrElse(t._2.id, "unknown"), metaUri, branch.parentEntry.get).foreach(_()))
+      }
+      case _ =>
+    }
+  }
+}
+
+object PatchedHover {
+
+  def apply(provider: VocabularyProvider,
+            seq: Seq[(BaseUnit, Dialect)],
+            branch: YPartBranch,
+            obj: AmfObject): PatchedHover = {
+    val ph = new PatchedHover(provider)
+    ph.init(seq, branch, obj: AmfObject)
```

Comment:
why is `branch` and `obj` necessary on `init` (the whole class scope)?, position-based parameters should only be on the scope of the getter

Reply:
removed from init